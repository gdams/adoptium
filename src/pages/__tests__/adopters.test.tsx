import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest'
import { axe } from 'vitest-axe';
import Adopters, { Head } from '../adopters';

vi.mock('../../util/shuffle', () => {
  return {
    shuffle: (array) => {
      array = [
        {
          name: 'mock_member',
          logo: 'mock_logo.png',
          url: 'https://mock.com',
          tier: 'mock_tier',
        }
      ]
      return array
    }
  };
});

describe('Adopters page', () => {
  it('renders correctly', () => {
    const { container } = render(<Adopters />);

    // eslint-disable-next-line
    const pageContent = container.querySelector('main');

    expect(pageContent).toMatchSnapshot();
  });

  it('head renders correctly', () => {
    const { container } = render(<Head />);
    // eslint-disable-next-line
    const title = container.querySelector('title');
    expect(title?.textContent).toEqual('Eclipse Temurin Adopters | Adoptium');
  });

  it('has no accessibility violations', async () => {
    const { container } = render(<Adopters />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
