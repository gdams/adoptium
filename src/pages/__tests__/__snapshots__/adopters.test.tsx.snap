// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Adopters page > renders correctly 1`] = `
<main>
  <section
    class="py-5 text-center container"
    style="max-width: 80em;"
  >
    <div
      class="row py-lg-5"
    >
      <div
        class="col-lg-8 col-md-8 mx-auto"
      >
        <h1
          class="fw-light"
        >
          Eclipse Temurin™ Adopters
        </h1>
        <p
          class="lead text-muted"
        >
          Companies that use Eclipse Temurin in production.
        </p>
        <div
          class="container"
        >
          <div
            class="row justify-content-center align-items-center"
          >
            <div
              class="col-lg-3 col-md-12 mt-5"
            >
              <a
                href="https://mock.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="mock_member logo"
                  src="/images/mock_logo.png"
                  style="width: 10rem;"
                />
              </a>
            </div>
          </div>
        </div>
        <a
          aria-controls="getListed"
          aria-expanded="false"
          class="btn btn-secondary mt-5"
          data-bs-toggle="collapse"
          href="#getListed"
          role="button"
        >
          How Can I Get My Logo Displayed?
        </a>
        <div
          class="collapse"
          id="getListed"
        >
          <div
            class="card card-body text-start"
          >
            <p>
              You can easily add your organization’s logo to our list of adopters by creating an issue or by submitting a pull request.
            </p>
            <p
              class="h3"
            >
              Option 1 - Open an Issue
            </p>
            <ol>
              <li>
                Go to our
                 
                <a
                  href="https://github.com/adoptium/adoptium.net/issues/new?assignees=&labels=adopter&projects=&template=adopters.yml&title=%5BAdopter%5D%3A+"
                  rel="noreferrer"
                  target="_blank"
                >
                  Temurin Adopters Form
                </a>
                .
              </li>
              <li>
                Fill out the relevant fields (it takes less than 5 minutes)
              </li>
              <li>
                Attach logo files to an issue by dragging and dropping them in the text editor of the form.
              </li>
            </ol>
            <p
              class="h3"
            >
              Option 2 - Submit a Pull Request
            </p>
            <p>
              When submitting a pull request, please make the following changes to the eclipsefdn-project-adopters'
               
              <a
                href="https://gitlab.eclipse.org/eclipsefdn/it/api/eclipsefdn-project-adopters"
                rel="noreferrer"
                target="_blank"
              >
                codebase
              </a>
              :
            </p>
            <ol>
              <li>
                Go to
                 
                <a
                  href="https://gitlab.eclipse.org/eclipsefdn/it/api/eclipsefdn-project-adopters/"
                  rel="noreferrer"
                  target="_blank"
                >
                  https://gitlab.eclipse.org/eclipsefdn/it/api/eclipsefdn-project-adopters
                </a>
                .
              </li>
              <li>
                Fork the repository.
              </li>
              <li>
                Update the adopter data file 
                <code>
                  config/adopters.json
                </code>
                . If your organization supports multiple projects, another project can be added to the projects list within the organization’s node. The values in this list should be the ID of the project.
              </li>
              <li>
                Add a colored and a white organization logo to
                 
                <code>
                  static/assets/images/adopters
                </code>
                .
              </li>
              <li>
                Submit the pull request.
              </li>
            </ol>
          </div>
        </div>
      </div>
    </div>
  </section>
</main>
`;
